Service/Worker/AbstractWorker.php

```diff
@@ -56,4 +65,39 @@ protected function postExecute(Message $message)
      * @return boolean
      */
     abstract protected function execute(Message $message);
+
+    /**
+     * Event fired when worker has processed message successfully.
+     *
+     * @return void
+     */
+    abstract protected function onSucceeded();
```

Comment:
Dear @petehouston 

I don't think we should mark this method as abstract. It's an optional and the developers could decide to handle that or not. Let you mark it as empty method.